vue-components/tests/e2e/specs/Dialog.test.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Gets the WebElement ID from the result of a `client.activeElement` call.
+ * In chrome,  result.value always has an `ELEMENT` key, whereas in other browser it
+ * has a different format with varying keys. This function assumes that `result.value`
+ * is always an object with a single entry.
+ *
+ * @param {Object} result
+ * @return {string}
+ */
+function getWebElementIdFromActiveElementResult( result ) {
+	return Object.values( result.value )[ 0 ];
+}
+
+describe( 'Dialog', function () {
+
+	beforeEach( function ( client ) {
+		client
+			.pause( 500 )
+			.init( client.launch_url + '/iframe.html?id=dialog--complex&viewMode=story' );
+	} );
+
+	afterEach( function ( client, done ) {
+		client.sauceEnd();// custom command used to finish the test if running in SauceLabs
+		client.end();
+		done();
+	} );
+
+	it( 'traps page focus, so that only visually focused elements are tab-able', function ( client ) {
+
+		const currentBrowser = client.options.desiredCapabilities.browserName;
+
+		// exclude internet explorer for now because there is a known bug in the .keys
+		// and .sendKeys functions
+		// ie: https://github.com/nightwatchjs/nightwatch/issues/2536
+		// firefox: https://github.com/nightwatchjs/nightwatch/issues/2468
+		if ( currentBrowser !== 'internet explorer' && currentBrowser !== 'safari' ) {
```

Comment:
If possible I'd prefer using [guard clauses](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) here instead of nesting the whole test, i.e.
```
if ( currentBrowser === 'internet explorer' || currentBrowser === 'safari' ) {
  return;
}
// actual test
```

Even cleaner could probably be using [test tags](https://nightwatchjs.org/guide/running-tests/test-tags.html#using-test-tags) but if I understood it right this would only work per file, so that would probably not work nicely.

Reply:
I agree that test tags would probably not work in this case. At least not as shown in the linked documentation.

Reply:
addressed